### ISSUE TITLE
Hang at specific FEN position

### ISSUE THREAD

**Issue by b9f9fe66** [official-stockfish/Stockfish#2091]
`q1B5/1P1q4/8/8/8/6R1/8/1K1k4 w - - 0 1`
[backtrace.txt](https://0x0.st/zNoN.txt)
Seems to be an infinite loop at:
https://github.com/official-stockfish/Stockfish/blob/5b5687d76e7dc10af9ebaa91f03190db9a9f8d27/src/search.cpp#L1082

**Comment by ddugovic** [comment]
Hang also reproducible with:

```
position fen q1B5/1P1q4/8/8/8/6R1/8/1K1k4 w - - 0 1
go depth 13
```

**Comment by mcostalba** [comment]
@b9f9fe66  @ddugovic thanks for reporting. BNug is confirmed and fixed here 5928cb2b300baafb04
